src/aichat/data/graphutil.js

```diff
@@ -73,8 +73,43 @@ function draw_force_directed_graph(graph, width, height, tag, process_group, pro
         .enter().append("line")
```

Comment:
I have a lot of suggestions for the javascript code, but I think that a good place to begin would be to simply understand some JS patterns. 

To start, try to convert this file (and all JS files) to the module pattern. If you don't you're bound to get conflicts since everything is in the global namespace. 

See here: https://addyosmani.com/resources/essentialjsdesignpatterns/book/#modulepatternjavascript